core/server/proxy/src/main/java/alluxio/proxy/s3/S3AuthenticationFilter.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.proxy.s3;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Collections;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerRequestFilter;
+import javax.ws.rs.container.PreMatching;
+import javax.ws.rs.ext.Provider;
+
+/**
+ * Authentication filter for S3 API.
+ * It will convert the content of the Authorization Header field to the user name.
+ */
+@PreMatching
+@Provider
+public class S3AuthenticationFilter implements ContainerRequestFilter {
+  private static final Logger LOG = LoggerFactory.getLogger(S3AuthenticationFilter.class);
+
+  @Override
+  public void filter(ContainerRequestContext requestContext) throws IOException {
+    if (!requestContext.getUriInfo().getPath().startsWith(S3RestServiceHandler.SERVICE_PREFIX)) {
+      return;
+    }
+    try {
+      String authorization = requestContext.getHeaderString("Authorization");
+      String user = S3RestUtils.getUser(authorization, requestContext);
+      // replace the authorization header value to user
+      requestContext.getHeaders().replace("Authorization", Collections.singletonList(user));
```

Comment:
Instead of replacing the "Authorization" header instead write the parsed user under a custom Alluxio header (eg: "Alluxio Fs User: <username>") and parse that in `S3RestServiceHandler` instead of the "Authorization" header.
- You might need to refactor the JAX-RS matching parameters in `S3RestServiceHandler`

Reply:
can you take a look at the latest commit? the change is about this question.